rocketmq-transport-binding.md

```diff
@@ -0,0 +1,231 @@
+contentType attribute.
+
+#### 3.1.2. Event Data Encoding
+
+The data attribute byte-sequence MUST be used as the value of the RocketMQ message.
+
+#### 3.1.3. Metadata Headers
+
+All CloudEvents attributes and CloudEvent Attributes Extensions with exception of data MUST be individually mapped to and from the Header fields in the RocketMQ message.
+
+##### 3.1.3.1 Property Names
+
+[CloudEvents][CE] attributes are prefixed with `"CE_"` for use in the message section.
+
+Examples:
+
+    * `eventTime` maps to `CE_eventTime`
+    * `eventID` maps to `CE_eventID`
+    * `cloudEventsVersion` maps to `CE_cloudEventsVersion`
+
+##### 3.1.3.2 Property Values
+
+The value for each RocketMQ Message header is constructed from the respective header's RocketMQ 
+representation, compliant with the RocketMQ message format specification.
+
+#### 3.1.4 Example
+
+This example shows the binary mode mapping of an event into the RocketMQ message. 
+All other CloudEvents attributes are mapped to RocketMQ message property fields with prefix `CE_`.
+
+Mind that `CE_` here does refer to the event data content carried in the payload.
+
+``` text
+------------------ Message -------------------
+
+Topic: mytopic
+
+
+-------------- user properties ---------------
+
+CE_contentType: application/avro
+CE_cloudEventsVersion: "0.1"
+CE_eventType: "com.example.someevent"
+CE_eventTime: "2018-11-23T03:56:24Z"
+CE_eventID: "1234-1234-1234"
+CE_source: "/mycontext/subcontext"
+       .... further attributes ...
+
+------------------- value --------------------
+
+            ... application data ...
+
+-----------------------------------------------
+```
+
+### 3.2. Structured Content Mode
+
+The [structured content mode](#32-structured-content-mode) keeps event metadata and data together in the payload, 
+allowing simple forwarding of the same event across multiple routing hops, and across 
+multiple transports.
+
+#### 3.2.1. RocketMQ Content-Type
+
+The [RocketMQ][RocketMQ] `CE_contentType` property field MUST be set to the media type of an event format.
+
+Example for the JSON format:
+
+```
+content-type: application/cloudevents+json; charset=UTF-8
+
+```
+
+#### 3.2.2. Event Data Encoding
+The chosen event format defines how all attributes, including the payload, are represented. 
+And in RocketMQ Message Header, it describes what is the type of transport event.
+
+The event metadata and data MAY then be rendered in accordance with the event format specification 
+and the resulting data becomes the payload.
+
+#### 3.2.3. Metadata Headers
+
+Implementations MAY include the same RocketMQ headers as defined for the binary mode.
+
+#### 3.2.4. Example
+This example shows a JSON event format encoded structured data event:
+
+``` text
+------------------ Message -------------------
+
+Topic: mytopic
+
+------------------ user properties -------------------
+
+CE_contentType: application/cloudevents+json; charset=UTF-8
```

Comment:
Isn't there some RocketMQ defined header that should hold this value?

Reply:
Year,I think Clound Event properties value hold this values more suitable rather than RocketMQ defined header.